src/views/app/Workforce-management/create-user.tsx

```diff
@@ -4,102 +4,157 @@ import ViewWrapper from '../../../components/layout/viewWrapper';
+import { CreateUserSchema } from '../../../utils/schemas';
 import { Icon } from '@rneui/themed';
 import { THEME_CONFIG } from '../../../config/appConfig';
 import { NativeStackScreenProps } from '@react-navigation/native-stack';
 import { InputComponent } from '../../../components/atoms/input';
+import { useGetRolesQuery } from '../../../store/services/role';
 
-const createUser: React.FC<NativeStackScreenProps<ParamListBase>> = props => {
-    const { goBack, setOptions } = useNavigation();
+const CreateUser: React.FC<NativeStackScreenProps<ParamListBase>> = props => {
+    const { navigation } = props;
+    const { goBack } = navigation;
 
     const {
-        user: { campus },
+        user: { campus, userId },
     } = useRole();
 
-    const [icon, setIcon] = React.useState<{ name: string; type: string }>({
-        type: 'ionicon',
-        name: 'briefcase-outline',
-    });
-
     const [departmentId, setDepartmentId] = React.useState<IDepartment['_id']>(); //Just for 3P testing
+    const [roleId, setRoleId] = React.useState<IRole['_id']>();
 
     const { setModalState } = useModal();
 
     const {
         data: campusDepartments,
         refetch: refetchDepartments,
-        isSuccess: campusDepartmentsSuccess,
+        isFetching: isFetchingDepartments,
         isLoading: campusDepartmentsLoading,
     } = useGetDepartmentsByCampusIdQuery(campus._id);
 
+    const {
+        data: allRoles,
+        refetch: refetchRoles,
+        isFetching: isFetchingRoles,
+        isLoading: rolesIsLoading,
+    } = useGetRolesQuery();
+
     const {
         data: workers,
-        isLoading: workersLoading,
-        isSuccess: workerSuccess,
+        refetch: refetchWorkers,
+        isFetching: isFetchingWorkers,
     } = useGetUsersByDepartmentIdQuery(departmentId as string, {
         skip: !departmentId,
     });
 
-    const { data: departments, isError: departmentsError } = useGetDepartmentsQuery();
+    const [uploadUser, { isError, isLoading, isSuccess, error }] = useUploadUserMutation();
 
-    const [createUser, { isError, isLoading, isSuccess, error }] = useCreateUserMutation();
+    const submitForm: FormikConfig<ICreateUserPayload>['onSubmit'] = (values, { resetForm }) => {
+        let stop = false;
+        const newValues = {
+            ...values,
+            departmentId,
+            roleId,
+            isRegistered: false,
+        };
 
-    const handleSubmit: FormikConfig<IRegisterPayload>['onSubmit'] = (values, { resetForm }) => {
-        createUser(values);
-        resetForm(INITIAL_VALUES);
-        setDepartmentId('');
-    };
+        Object.entries(newValues).forEach(array => {
+            if (array[1] === undefined) {
+                setModalState({
+                    message: `Plese select a ${array[0]}`,
+                    defaultRender: true,
+                    status: 'error',
+                    duration: 3,
+                });
+                stop = true;
+            }
+        });
```

Comment:
If your validation schema is updated with all the required values, then you won't need this.